submit/controllers/ui/new/metadata.py

```diff
@@ -155,26 +148,25 @@ def optional(method: str, params: MultiDict, session: Session,
         'submission': submission
     }
 
-    if method == 'POST':
-        if not form.validate():
-            raise BadRequest(response_data)
-
+    if method == 'POST' and form.validate():
         logger.debug('Form is valid, with data: %s', str(form.data))
 
         commands, valid = _opt_commands(form, submission, submitter, client)
         # We only want to apply updates if the metadata has actually changed.
-        if commands:   # Metadata has changed.
-            if not all(valid):
-                raise BadRequest(response_data)
-
+        if not commands:
+            return ready_for_next((response_data, status.OK, {}))
+        if all(valid):  # Metadata has changed and is all valid
             try:
                 submission, _ = save(*commands, submission_id=submission_id)
+                response_data['submission'] = submission
+                return ready_for_next((response_data, status.OK, {}))
             except SaveError as e:
                 raise InternalServerError(response_data) from e
-            response_data['submission'] = submission
+        else:
+            stay_on_this_stage((response_data, status.OK, {}))
+    else:
+        stay_on_this_stage((response_data, status.OK, {}))
```

Comment:
These lines could be removed, if L170 is changed to `stay_on_this_stage((response_data, status.OK, {}))`.

Reply:
:+1: